File: Backend_Draft/driver-service/src/main/java/com/swiftmove/driverservice/model/Vehicle.java
package com.swiftmove.driverservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "vehicle")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "model", nullable = false)
    private String model;

    @Column(name = "make", nullable = false)
    private String make;

    @Column(name = "year", nullable = false)
    private Integer year;

    @Column(name = "color", nullable = false)
    private String color;

    @Column(name = "price_per_km", nullable = false)
    private Long pricePerKm;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @Column(name = "can_carry_furniture", nullable = false)
    private Boolean canCarryFurniture;

    @Column(name = "driver_id", nullable = false)
    private Long driverInfoId;

    @Column(name = "vehicle_type_id", nullable = false)
    private Long vehicleTypeId;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private java.sql.Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private java.sql.Timestamp updatedAt;
}

File: src/swiftmove-dashboards/src/types/old-frontend/vehicle.ts
export type VehicleTypeEnum = 'SEDAN' | 'SUV' | 'HATCHBACK' | 'MINIVAN' | 'VAN' | 'TRUCK';

export interface VehicleType {
  id?: string | number;
  type: VehicleTypeEnum;
  volume: number; // cubic feet
  maxWeight: number; // lbs
}

export interface Vehicle {
  id: string | number;
  model: string;
  make: string;
  year: number;
  color: string;
  licensePlate?: string;
  pricePerKm: number;
  isActive: boolean;
  canCarryFurniture: boolean;
  driverInfoId: string | number;
  vehicleTypeId: string | number;
  vehicleType?: VehicleType;
  createdAt?: string;
  updatedAt?: string;
}

export interface VehicleFormData {
  model: string;
  make: string;
  year: number;
  color: string;
  licensePlate?: string;
  pricePerKm: number;
  canCarryFurniture: boolean;
  vehicleTypeId: string | number;
}
